opm/input/eclipse/Schedule/MSW/Segment.cpp

```diff
@@ -96,6 +98,15 @@ namespace Opm {
         , m_x                (0.0)
         , m_y                (0.0)
     {
+        if (m_roughness != invalid_value && m_internal_diameter != invalid_value) {
+            const double safe_roughness = m_internal_diameter * std::min(MAX_REL_ROUGHNESS, m_roughness/m_internal_diameter);
+            if (safe_roughness != m_roughness) {
```

Comment:
equality comparison for floating point values are not reliable.  You can either do the comparison with a small value tolerance or a different way to determine that the value is too big and needs to be clapped.